### DISCUSSION TITLE
fulltextsearch platform seems down

### DISCUSSION THREAD

**Discussion by qudiqudi** [nextcloud/all-in-one#5064] [Questions] [answered]
These two errors are spamming my log every time I use the search:

```
fulltextsearch	
platform seems down. we will update index next cron tick 
```

```
fulltextsearch_elasticsearch	
Retry 0: cURL error 7: Failed to connect to nextcloud-aio-fulltextsearch port 9200 after 56 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://elastic:{token_redacted}@nextcloud-aio-fulltextsearch:9200/nextcloud-aio/_doc/files%3A829880
```

I already did the "remove and add master container to nextcloud-aio" - tip and read through [this discussion](https://github.com/nextcloud/all-in-one/discussions/4290), unfortunately nothing there is helping.
I can see that the nextcloud-aio-fulltextsearch container is part of the nextcloud-aio network and has an ipv4 address.
curl from inside the fulltextsearch container is able to connect and retrieve information using:

`curl -XGET 'http://elastic:{redacted}@nextcloud-aio-fulltextsearch:9200/nextcloud-aio/_doc/files%3A829880'`

So this leads me to suspect there is still some issue with docker networking, as I can't curl the same from inside the nextcloud-aio-nextcloud container:

`curl: (7) Failed to connect to nextcloud-aio-fulltextsearch port 9200 after 59 ms: Couldn't connect to server`

But I can ping both the ip and container name from within nextcloud-aio-nextcloud, so it can't be any dns related issue.

```
3327f283c658:/var/www/html# ping nextcloud-aio-fulltextsearch
PING nextcloud-aio-fulltextsearch (172.17.0.2): 56 data bytes
64 bytes from 172.17.0.2: seq=0 ttl=42 time=0.105 ms
64 bytes from 172.17.0.2: seq=1 ttl=42 time=0.069 ms
64 bytes from 172.17.0.2: seq=2 ttl=42 time=0.069 ms
```

Any other pointers? Thanks!

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker info` here?

**Reply by qudiqudi** [reply]
Sure, here you go:

```
sudo docker info
Client: Docker Engine - Community
 Version:    27.1.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 11
  Running: 9
  Paused: 0
  Stopped: 2
 Images: 11
 Server Version: 27.1.1
 Storage Driver: overlay2
  Backing Filesystem: zfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2bf793ef6dc9a18e00cb12efb64355c2c9d5eb41
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.6.32-production+truenas
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 62.59GiB
 Name: nextcloud
 ID: LDD6:UC6F:EYZO:PU33:WQLI:ELJE:ZO2G:XTGN:GLIZ:HCKL:MAIN:UQ3B
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Reply by szaimen** [reply]
Can you reset the fts index and check if that helps to make it work? https://github.com/nextcloud/all-in-one/discussions/2509

**Reply by qudiqudi** [reply]
Same problem like here: https://github.com/nextcloud/fulltextsearch/issues/831

```
WARNING! This operation is not reversible.
Please confirm this destructive operation by typing 'reset ALL ALL': reset ALL ALL

In SimpleNodePool.php line 77:

  No alive nodes. All the 1 nodes seem to be down.
```

Proceeding with removing the fts index does nothing, unfortunately.

Also, the user password doesn't consists of special characters, for that matter.

**Reply by szaimen** [reply, marked as answer]
Yeah this seems to be a bug in the fts app if one cannot reset the index anymore. Please subscribe to https://github.com/nextcloud/fulltextsearch/issues/831

**Reply by qudiqudi** [reply]
@szaimen, as it seems, it can be fixed with a small change to the FULLTEXTSEARCH_HOST variable, derived from a curl bug. See here: https://github.com/nextcloud/fulltextsearch/issues/831#issuecomment-2344795990
